clash-lib/src/Clash/Netlist/Id.hs

```diff
@@ -242,5 +242,8 @@ fromCoreId = withIdentifierSetM fromCoreId#
 -- spliced at verbatim in the HDL. As opposed to 'fromCoreId', the resulting
 -- Identifier might be generated at a later point as it is NOT added to an
 -- IdentifierSet.
-unsafeFromCoreId :: HasCallStack => Id -> Identifier
-unsafeFromCoreId = unsafeMake . nameOcc . varName
+unsafeFromCoreId :: HasCallStack => SanitizeNames -> Id -> Identifier
+unsafeFromCoreId san = unsafeMake . ckId . nameOcc . varName where
+  ckId s = case san of
+    Sanitize -> Data.Text.map (\c -> if c == '$' then '_' else c) s
+    NoSanitize -> s
```

Comment:
`ckId` isn't a particularly meaningful name. A better name might be `maybeSanitize`

Reply:
Also done!